File: Algorithms/Precompute/Prefix Sums/subarraySum.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        sum = 0
        counter = {}
        counter[0] = 1
        ans = 0
        for n in nums:
            sum += n
            if sum-k in counter: ans += counter[sum-k]
            if sum not in counter: counter[sum] = 0
            counter[sum] += 1

        return ans

File: Algorithms/Precompute/Prefix Sums/subarraySum.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        std::unordered_map<int, int> map;
        map[0] = 1;
        int sum = 0;
        int count = 0; 
        for(auto n: nums){
            sum += n; 
            if(map.count(sum-k) > 0) count += map[sum-k];
            if(map.count(sum) == 0) map[sum] = 0;
            map[sum]++;
        }
        
        return count;
    }
};